### PR TITLE
algebra: adding tests for `vec_cross` function

### PR DESCRIPTION
**Pull request by PNieck**
JIRA: PP-71

## Description

This PR contains tests for `vec_cross` functions

## Motivation and Context

More precise testing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by niewim19 — COMMENTED** [review]
Current tests are good to go. 

However, cross product of 3d vectors has its edge cases of cross product of two perpendicular vectors (both directions). Please add a test for it (i think both cases cand be checked in one test)

File: algebra/tests/vec/various.c [hunk 1]
```diff
@@ -303,3 +310,103 @@ TEST_GROUP_RUNNER(group_vec_sub)
...
+}
+
+
+TEST(group_vec_cross, vec_cross_parallel)
+{
+	vec_t A = V2;
+	vec_t B = A;
+	vec_t C;
+
+	vec_times(&B, POS_SCALAR);
```
Comment (niewim19) [inline algebra/tests/vec/various.c:387]: This test do two things, better to pinpoint what happens now :)
```suggestion

	/* Parallel with common direction */
	vec_times(&B, POS_SCALAR);
```

File: algebra/tests/vec/various.c [hunk 2]
```diff
@@ -303,3 +310,103 @@ TEST_GROUP_RUNNER(group_vec_sub)
...
+
+	vec_cross(&A, &B, &C);
+
+	TEST_ASSERT_EQUAL_FLOAT(0.0, C.x);
+	TEST_ASSERT_EQUAL_FLOAT(0.0, C.y);
+	TEST_ASSERT_EQUAL_FLOAT(0.0, C.z);
+
+	B = A;
+	vec_times(&B, NEG_SCALAR);
```
Comment (niewim19) [inline algebra/tests/vec/various.c:396]:
```suggestion

	/* Parallel with opposite directions */
	B = A;
	vec_times(&B, NEG_SCALAR);
```

**Review by niewim19 — APPROVED** [review]
Tests look good.

Maybe comments would be good in parallel checks, but it is not a merge blocker

File: algebra/tests/vec/various.c (hunk 1 above)
Comment (PNieck) [reply algebra/tests/vec/various.c]: Done

**Review by niewim19 — APPROVED** [review]
Great, LGTM
